contribution_to_project.rst

```diff
@@ -0,0 +1,95 @@
+  - :ref:`Contribution to user experience<Contribution to user exp>`.
+  - :ref:`Contribution to a community<Contribution to community>`.
+  - :ref:`Other kinds of contributions<Other kinds of contributions>`.
+
+.. _Contribution to codebase:
+
+Contribution to a codebase
+--------------------------
+
+Contribution to software itself (its codebase) insludes:
+
+  - Proposing new features.
+  - Reporting bugs.
+  - Participating in design and implementation of new components.
+  - Bug fixes.
+  - Code reviewing.
+  - Code refactoring.
+  - Improving stability of an existing codebase through improving a testing codebase and infrastructure.
+  - Maintenance (for example, backporting and releasing).
+
+Examples of codebase contributions which do not require programming skills in scope of the Ansible project can be reviewing proposals, testing pull requests locally, adding integration tests, bug reports, feature requests, providing user experience to programmers in issues and pull requests.
+
+.. _Contribution to dev exp:
+
+Contribution to code developer experience
+-----------------------------------------
+
+Contribution to code developer experience includes sharing experience through improving code development documentation, writing development guidelines, mentoring and training code contributors.
+
+.. _Contribution to user exp:
+
+Contribution to user experience
+-------------------------------
+
+Contribution to user experience includes sharing experience about how to use software more efficiently through improving user documentation, writing manuals and books, making videos, courses, and so on.
+
+Examples:
+
+  - Clarification and updating user documentation.
+  - Making a presentation for a local meetup covering not well-known but useful usage aspects.
+
+.. _Contribution to community:
+
+Contribution to a community
+-----------------------------
+
+In a narrow sense, contribution to a community includes:
+
+  - Promoting the project among people.
+  - Thinking of and proposing ideas about how community processes can be improved.
+  
+Examples:
+
+  - Organizing a meetup.
+  - Writing a blogpost about benefits of the project comparing with other similar projects.
+  - Making a presentation on another IT community's event showing how the project can be used by that community to solve their issues.
+  - Suggesting ideas on how the process of decision making can be improved.
+
+In a broader sense, contribution to a community includes the activities mentioned in all the other types of contribution when they imply good interaction between participants or good impact on the community health and growth.
+
+Examples:
+
+  - If you write a guide which significantly reduces a threshold of the first patch, it can lead to increase of a number of code contributors, which is definitely a contribution to the community.
+  - Active participation in IRC / mail / other discussions.
```

Comment:
Or
```suggestion
  - Active participation using IRC / mail / other methods.
```

Reply:
I'd like to highlight the discussions here. It implies even in-person discussions in meetups and any other kinds of discussions. So i'd leave `discussions` here:)